src/icalendar/cal.py

```diff
@@ -390,14 +390,22 @@ def from_ical(cls, st, multiple=False):
         if multiple:
             return comps
         if len(comps) > 1:
-            raise ValueError(f'Found multiple components where '
-                             f'only one is allowed: {st!r}')
+            raise ValueError(cls._format_error(
+                'Found multiple components where only one is allowed', st))
         if len(comps) < 1:
-            raise ValueError(f'Found no components where '
-                             f'exactly one is required: '
-                             f'{st!r}')
+            raise ValueError(cls._format_error(
+                'Found no components where exactly one is required', st))
         return comps[0]
 
+    def _format_error(error_description, bad_input, elipsis='[...]'):
+        # there's three character more in the error, ie. ' ' x2 and a ':'
+        max_error_length = 100 - 3
```

Comment:
This is a code smell called magic number. This number would be in a constant e.g. `MAXIMUM_ERROR_LENGTH`, best documented with a comment.